### ISSUE TITLE
Fix false positives for dollar variables in function-linear-gradient-no-nonstandard-direction

### ISSUE THREAD

**Issue by ccabrales** [stylelint/stylelint#4014]
> Clearly describe the bug

There are false positives for nonstandard directions when using an SCSS variable with one of the directions (e.g. `top`) in its name.

> Which rule, if any, is the bug related to?

`function-linear-gradient-no-nonstandard-direction`

> What CSS is needed to reproduce the bug?

```css
.foo {
  background: linear-gradient($purple-top, $purple-bottom);
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-linear-gradient-no-nonstandard-direction": true
  }
}
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint --config stylelint.config.js "src/**/*.scss" -f verbose`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, related to SCSS variables.

> What did you expect to happen?

No errors to be flagged on the usage of `linear-gradient`.

> What actually happened (e.g. what warnings or errors did you get)?

There is an error because of the variable name.

**Comment by jeddy3** [comment]
@ccabrales Thanks for the report and for using the template.

I can reproduce this. The fix should just be the case of returning early if the first value in the function isn't [standard syntax](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxValue.js).

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).
